railties/test/generators/app_generator_test.rb

```diff
@@ -205,149 +202,128 @@ def test_app_update_create_new_framework_defaults
+    stub_rails_application do
+      generator = Rails::Generators::AppGenerator.new ["rails"], { update: true, asset_pipeline: "propshaft" }, { destination_root: destination_root, shell: @shell }
       generator.send(:app_const)
       quietly { generator.update_config_files }
 
-      assert_file "#{app_root}/config/initializers/assets.rb"
-      assert_no_file "#{app_root}/app/assets/config/manifest.js"
+      assert_file "config/initializers/assets.rb"
+      assert_no_file "app/assets/config/manifest.js"
     end
   end
 
   def test_app_update_does_not_generate_action_cable_contents_when_skip_action_cable_is_given
-    app_root = File.join(destination_root, "myapp")
-    run_generator [app_root, "--skip-action-cable"]
+    run_generator [destination_root, "--skip-action-cable"]
 
-    stub_rails_application(app_root) do
-      generator = Rails::Generators::AppGenerator.new ["rails"], { update: true, skip_action_cable: true }, { destination_root: app_root, shell: @shell }
+    stub_rails_application do
+      generator = Rails::Generators::AppGenerator.new ["rails"], { update: true, skip_action_cable: true }, { destination_root: destination_root, shell: @shell }
       generator.send(:app_const)
       quietly { generator.update_config_files }
 
-      assert_no_file "#{app_root}/config/cable.yml"
-      assert_file "#{app_root}/config/environments/production.rb" do |content|
+      assert_no_file "config/cable.yml"
+      assert_file "config/environments/production.rb" do |content|
         assert_no_match(/config\.action_cable/, content)
       end
-      assert_no_file "#{app_root}/test/channels/application_cable/connection_test.rb"
+      assert_no_file "test/channels/application_cable/connection_test.rb"
     end
   end
 
   def test_app_update_does_not_generate_bootsnap_contents_when_skip_bootsnap_is_given
-    app_root = File.join(destination_root, "myapp")
-    run_generator [app_root, "--skip-bootsnap"]
+    run_generator [destination_root, "--skip-bootsnap"]
 
-    stub_rails_application(app_root) do
-      generator = Rails::Generators::AppGenerator.new ["rails"], { update: true, skip_bootsnap: true }, { destination_root: app_root, shell: @shell }
+    stub_rails_application do
+      generator = Rails::Generators::AppGenerator.new ["rails"], { update: true, skip_bootsnap: true }, { destination_root: destination_root, shell: @shell }
       generator.send(:app_const)
       quietly { generator.update_config_files }
 
-      assert_file "#{app_root}/config/boot.rb" do |content|
+      assert_file "config/boot.rb" do |content|
         assert_no_match(/require "bootsnap\/setup"/, content)
       end
     end
   end
 
   def test_app_update_preserves_skip_active_job
-    app_root = File.join(destination_root, "myapp")
-    run_generator [ app_root, "--skip-active-job" ]
+    run_generator [ destination_root, "--skip-active-job" ]
 
-    FileUtils.cd(app_root) do
```

Comment:
This test and the others like it are passing because the generator chdirs to its `destination_root`, rather than because `setup` does[^1].  It would probably be better not to rely on that though.

If we want to clean up the `cd` blocks a bit, we could introduce an `in_root` helper method, like generators themselves have.

Alternatively, if we want to eliminate the `cd` blocks, we could introduce `run_app_update` and `read_file` helper methods (and maybe `write_file` for [`test_app_update_does_not_change_config_target_version`](https://github.com/rails/rails/blob/1755c5ac28a0baf10df305573569f71babdf7545/railties/test/generators/app_generator_test.rb#L452)).  `run_app_update` would chdir as necessary, and `read_file` would simply `File.expand_path` with `destination_root`.  That could be a follow-up PR, though.

[^1]: [`setup`](https://github.com/rails/rails/blob/1755c5ac28a0baf10df305573569f71babdf7545/railties/lib/rails/generators/testing/setup_and_teardown.rb#L7-L11) actually calls [`ensure_current_path`](https://github.com/rails/rails/blob/1755c5ac28a0baf10df305573569f71babdf7545/railties/lib/rails/generators/testing/behavior.rb#L94-L96), which chdirs to [`File.expand_path(Dir.pwd)`](https://github.com/rails/rails/blob/1755c5ac28a0baf10df305573569f71babdf7545/railties/lib/rails/generators/testing/behavior.rb#L21).